check_api/src/main/java/com/google/errorprone/ErrorPronePlugins.java

```diff
@@ -45,11 +54,50 @@ public static ScannerSupplier loadPlugins(ScannerSupplier scannerSupplier, Conte
       return scannerSupplier;
     }
     ClassLoader loader = fileManager.getClassLoader(StandardLocation.ANNOTATION_PROCESSOR_PATH);
-    Iterable<BugChecker> extraBugCheckers = ServiceLoader.load(BugChecker.class, loader);
-    if (Iterables.isEmpty(extraBugCheckers)) {
+
+    List<Class<? extends BugChecker>> bugCheckerClasses;
+    // Synchronize to prevent races
+    synchronized(ErrorPronePlugins.class) {
+      // disable URL caching to avoid stale state bugs when run within a daemon-building process
+      boolean oldURLCaching = setUrlCaching(false);
```

Comment:
Can you structure this section as a try-finally?

get old value 
try {
  set to false;
  do work;
} finally {
  set old value;
}